server/src/main/java/org/elasticsearch/search/vectors/RescoreVectorBuilder.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.search.vectors;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class RescoreVectorBuilder implements Writeable, ToXContentObject {
+
+    public static final ParseField OVERSAMPLE_FIELD = new ParseField("oversample");
+    public static final float MIN_OVERSAMPLE = 1.0F;
+    private static final ConstructingObjectParser<RescoreVectorBuilder, Void> PARSER = new ConstructingObjectParser<>(
+        "rescore",
+        args -> new RescoreVectorBuilder((Float) args[0])
+    );
+
+    static {
+        PARSER.declareFloat(ConstructingObjectParser.optionalConstructorArg(), OVERSAMPLE_FIELD);
+    }
+
+    // Oversample is required as of now as it is the only field in the rescore vector
+    // that may change in the future, so we treat it as optional
+    private final Float oversample;
```

Comment:
Nah, lets treat it as mandatory and use a raw `float`

Reply:
OK!  My concern was for this not to be mandatory in the future, as this provided more BwC room IMO. Done in ddc60947ec76568e1ca2162b2cc956dc3528dc88